### PR TITLE
Fix Masking in Actor Logging

### PR DESCRIPTION
**Pull request by 4e6**
### Pull Request Description

`ActorLogging` trait overrides SLF4J logger preventing the masking.
PR replaces `ActorLoggin` usages with `LazyLogging` providing the SLF4J logger.

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/main/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/main/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/main/docs/style-guide/rust.md) style guides.
- [x] All documentation and configuration conforms to the [markdown](https://github.com/enso-org/enso/blob/main/docs/style-guide/markdown.md) and [YAML](https://github.com/enso-org/enso/blob/main/docs/style-guide/yaml.md) style guides.
- [x] All code has been tested where possible.

### CODE REVIEW

**Review by iamrecursion — APPROVED** [review]

File: engine/language-server/src/main/scala/org/enso/languageserver/monitoring/ReadinessMonitor.scala [hunk 1]
```diff
@@ -28,7 +29,7 @@ class ReadinessMonitor() extends Actor with ActorLogging {
       context.become(ready())
 
     case InitializationFailed =>
-      log.error("Initialization failed. Terminating JVM...")
+      logger.error("Initialization failed. Terminating JVM...")
       System.exit(1)
```
Comment (radeusgd) [inline engine/language-server/src/main/scala/org/enso/languageserver/monitoring/ReadinessMonitor.scala:33]: While this is not related to the change, I only now noticed it and have a feeling that this message may not be logged. As the underlying logger is using a message queue on a different thread, it may take a short moment before it is actually written and it may not do it on time before `System.exit` terminates the application.

File: engine/language-server/src/main/scala/org/enso/languageserver/util/UnhandledLogging.scala [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
+import akka.event.Logging
 
-trait UnhandledLogging { this: Actor with ActorLogging =>
+trait UnhandledLogging { this: Actor =>
+
+  private val akkaLogger = Logging(context.system, this)
 
   override def unhandled(message: Any): Unit =
-    log.warning("Received unknown message: {}", message)
+    akkaLogger.warning("Received unknown message: {}", message)
```
Comment (radeusgd) [inline engine/language-server/src/main/scala/org/enso/languageserver/util/UnhandledLogging.scala]: Is this surely the right logger to use?

Because at first glance it seems to me like it may be the same one as in `ActorLogging` which we seem to need to avoid.

**Review by radeusgd — APPROVED** [review]
Looks good to me, just one question.

File: engine/language-server/src/main/scala/org/enso/languageserver/util/UnhandledLogging.scala (hunk 2 above)
Comment (4e6) [reply engine/language-server/src/main/scala/org/enso/languageserver/util/UnhandledLogging.scala]: Akka logger is used here to be able to control the logger verbosity. Since this is related to the Akka logic, this will be controlled by the Akka log level settings.
But you made me think if we can use the SLF4J logger here

Comment (radeusgd) [reply engine/language-server/src/main/scala/org/enso/languageserver/util/UnhandledLogging.scala]: From my understanding it was that the Akka logging will not handle masking correctly.

But your point also makes sense. Moreover - I think this `message` of type `Any` will not trigger the masking mechanism anyway, right? Then maybe it can stay as is, unless we are worried that the message may contain PII.

Comment (4e6) [reply engine/language-server/src/main/scala/org/enso/languageserver/util/UnhandledLogging.scala]: If the message was designed to be mask (extended the `ToLogString`, it will be masked. But I changed the function to log just the class to be extra safe, or if the message is extremely big.
